### PR TITLE
i82: Use custom Navigation object for navigation

### LINKED ISSUE #82: Use NavigationService for all navigation
**Is your feature request related to a problem? Please describe.**
If all navigation goes through the NavigationService it will be a lot easier to make any required changes down the track.  

**Describe the solution you'd like**
Have functions in the NavigationService that perform navigation and use Provider to access the service everywhere we do navigation.  

Questions 
- [x] @tamari-gray do you want the NavigationService to be named something else to make it clearer that it's more an odd cousin of a service than a service?

### PR DESCRIPTION
**Pull request by nickmeinhold**
fixes #82

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -73,10 +76,21 @@ class _MyAppState extends State<MyApp> {
...
+            '/create_account': (context) => CreateAccountScreen(),
+            '/sign_in': (context) => SignInScreen(),
+            // TODO: complete when database strategy for games has been finalised
+            // '/input_password': (context) => InputPasswordScreen(),
+          },
           home: StreamBuilder(
             stream: widget._authService.streamOfAuthState,
             builder: (context, snapshot) {
-              // TODO: check for snapshot error and send to navigation manager for display
+              if (snapshot.hasError) {
```
Comment (tamari-gray) [inline lib/main.dart]: 🙏 champion, i was ready to push that off until we start deploying 👯‍♂️

File: lib/screens/create_account.dart [hunk 2]
```diff
@@ -193,13 +193,7 @@ class _CreateAccountScreenState extends State<CreateAccountScreen> {
...
                               _formKey.currentState.reset();
                               _clearForm();
-                              Navigator.pushReplacement<SignInScreen,
-                                  CreateAccountScreen>(
-                                context,
-                                MaterialPageRoute<SignInScreen>(
-                                  builder: (context) => SignInScreen(),
-                                ),
-                              );
+                              context.read<Navigation>().switchToSignIn();
```
Comment (tamari-gray) [inline lib/screens/create_account.dart:196]: 😍 sho clean

File: lib/screens/sign_in.dart [hunk 3]
```diff
@@ -15,17 +15,21 @@ class SignInScreen extends StatefulWidget {
 
 class _SignInScreenState extends State<SignInScreen> {
   final _formKey = GlobalKey<FormState>();
+
+  Navigation _navigation;
```
Comment (tamari-gray) [inline lib/screens/sign_in.dart]: hi @nickmeinhold, are you wanting to use the navigation as a variable or did you forget to delete it?
im easy either way, thanks for the good work!

**Review by tamari-gray — CHANGES_REQUESTED** [review]
just the sign in widget needs some clarification (refer to comment)
other then that thanks so much for the work today!
Sorry for making you do some of the more tedious tickets with alot of files to sift through and update 🙏 
I know i say thank you eveytime but i really am gratefull for the help!

File: lib/screens/sign_in.dart (hunk 3 above)
Comment (nickmeinhold) [reply lib/screens/sign_in.dart]: Totally forgot, awesome spot, thanks!  Fixing now...

Comment (nickmeinhold) [reply lib/screens/sign_in.dart]: Fixed.

**Review by tamari-gray — APPROVED** [review]
amazing thanks @nickmeinhold! ✨✨

**Pull request merged**